src/analysis/special_functions/trigonometric/inverse.lean

```diff
@@ -281,8 +289,24 @@ by rw [arccos, sub_eq_iff_eq_add, ← sub_eq_iff_eq_add', div_two_sub_self, neg_
 lemma arccos_neg (x : ℝ) : arccos (-x) = π - arccos x :=
 by rw [← add_halves π, arccos, arcsin_neg, arccos, add_sub_assoc, sub_sub_self, sub_neg_eq_add]
 
-lemma sin_arccos {x : ℝ} (hx₁ : -1 ≤ x) (hx₂ : x ≤ 1) : sin (arccos x) = sqrt (1 - x ^ 2) :=
-by rw [arccos_eq_pi_div_two_sub_arcsin, sin_pi_div_two_sub, cos_arcsin hx₁ hx₂]
+lemma arccos_of_one_le {x : ℝ} (hx : 1 ≤ x) : arccos x = 0 :=
+by rw [arccos, arcsin_of_one_le hx, sub_self]
+
+lemma arccos_of_le_neg_one {x : ℝ} (hx : x ≤ -1) : arccos x = π :=
+by rw [arccos, arcsin_of_le_neg_one hx, sub_neg_eq_add, add_halves']
+
+lemma sin_arccos (x : ℝ) : sin (arccos x) = sqrt (1 - x ^ 2) :=
```

Comment:
Again I think worth adding a comment.

Reply:
Comments added in both places, and also updated the doc string for `arccos` to correct the statement of what the junk values actually are.